signac/contrib/job.py

```diff
@@ -825,7 +843,9 @@ def __exit__(self, err_type, err_value, tb):
 
     def __setstate__(self, state):
         self.__dict__.update(state)
-        self.statepoint._parent.jobs.append(self)
+        # We append to a list of jobs rather than replacing to support
+        # transparent id updates between shallow copies of a job.
+        self.statepoint._jobs.append(self)
```

Comment:
If using a WeakSet as described above:
```suggestion
        self.statepoint._jobs.add(self)
```